src/ImageBackingStore.h

```diff
@@ -129,4 +147,53 @@ class ImageBackingStore
     char m_foldername[MAX_FILE_PATH + 1];
 
     bool _internal_open(const char *filename);
+
+#if ENABLE_COW
```

Comment:
How would you feel about having the COW stuff in its own class, which would then be as a member in `ImageBackingStore`? I think it would clean things up, and shouldn't be too hard to do. I mean move the parts here to a new `COWStorage.cpp` or similar, and instead of `cow_write()` call `m_cow.write()` etc. in ImageBackingStore.